keras/src/quantizers/quantizers.py

```diff
@@ -127,6 +127,341 @@ def get_config(self):
         }
 
 
+def adjust_and_nudge(min_range, max_range, num_bits, narrow_range):
+    """Adjusts and nudges the quantization range for better accuracy."""
+    if num_bits < 2:
+        raise ValueError("num_bits must be >= 2")
+
+    n_steps = float(2**num_bits - 1)
+    if narrow_range:
+        n_steps -= 1.0
+
+    # Handle the case where min and max are too close
+    if abs(max_range - min_range) < 1e-10:
+        return min_range, max_range, 1.0
+
+    # Calculate the step size
+    step_size = (max_range - min_range) / n_steps
+
+    # Calculate the reciprocal of the step size
+    inv_step_size = 1.0 / step_size
+
+    # Round the reciprocal to get an integer
+    rounded_inv_step_size = ops.round(inv_step_size)
+
+    # Calculate the final step size
+    final_step_size = 1.0 / rounded_inv_step_size
+
+    # Calculate the quantized min/max values, ensuring accurate rounding
+    quantized_min = (
+        ops.round(min_range * rounded_inv_step_size) / rounded_inv_step_size
+    )
+    quantized_max = (
+        ops.round(max_range * rounded_inv_step_size) / rounded_inv_step_size
+    )
+
+    # Convert quantization limits to float
+    quant_min_float = float(quantized_min)
+    quant_max_float = float(quantized_max)
+
+    # Calculate the scale
+    nudged_scale = (max_range - min_range) / (quant_max_float - quant_min_float)
+
+    # Calculate zero point from min
+    zero_point_from_min = quant_min_float - min_range / nudged_scale
+
+    # Determine nudged zero point
+    if zero_point_from_min < quant_min_float:
```

Comment:
The if statement might not work in the XLA context. Could these lines be vectorized instead?